cloudinit/netinfo.py

```diff
@@ -57,8 +57,10 @@ def _netdev_info_iproute(ipaddr_out):
             devs[dev_name]["ipv6"].append(m.groupdict())
         elif "inet" in line:
             m = re.match(
-                r"\s+inet\s(?P<cidr4>\S+)(\sbrd\s(?P<bcast>\S+))?\sscope\s"
-                r"(?P<scope>\S+).*",
+                r"\s+inet\s(?P<cidr4>\S+)"
+                r"(\smetric\s(?P<metric>\d+))?"
```

Comment:
I did a little bit of manual testing with this regex and didn't spot any issues.

Reply:
The bigger this regex gets, the more it makes me question whether we should really be extending a fragile regex parsing of human-readable output of ip.  While I originally weighed in to the contrary because cloud-init is supported even more broadly than just Ubuntu, I think it would probably be preferable given that we attempt to try `ip -json` format first, and fallback to the legacy parsing if `-json` param is not present. This way most of the latest *nix distros with recent `ip`  tooling can benefit from simple parsing of machine-readable output and any older distributions will continue to parse correctly with the older `ip` tooling and output formats.